torch/ao/quantization/_pt2e/quantizer/x86_inductor_quantizer.py

```diff
@@ -0,0 +1,205 @@
+import torch
+import torch.nn.functional as F
+import copy
+import functools
+import itertools
+from .quantizer import (
+    OperatorConfig,
+    OperatorPatternType,
+    QuantizationConfig,
+    QuantizationSpec,
+    Quantizer,
+    QuantizationAnnotation,
+)
+from torch.ao.quantization._pt2e.quantizer.utils import (
+    get_input_act_qspec,
+    get_output_act_qspec,
+    get_weight_qspec,
+    get_bias_qspec,
+)
+from .qnnpack_quantizer import (
+    _is_annotated,
+)
+from torch.ao.quantization.observer import (
+    HistogramObserver,
+    PlaceholderObserver,
+    PerChannelMinMaxObserver,
+)
+from torch.ao.quantization.qconfig import _ObserverOrFakeQuantizeConstructor
+from typing import List, Dict, Optional, Set, Any
+from torch.fx import Node
+from torch.fx.passes.utils.source_matcher_utils import get_source_partitions
+
+__all__ = [
+    "X86InductorQuantizer",
+    "get_default_x86_inductor_quantization_config",
+]
+
+def supported_quantized_operators() -> Dict[str, List[OperatorPatternType]]:
+    supported_operators: Dict[str, List[OperatorPatternType]] = {
+        "conv2d": [
+            [torch.nn.Conv2d],
+            [F.conv2d],
+        ],
+    }
+    return copy.deepcopy(supported_operators)
+
+
+def get_supported_x86_inductor_config_and_operators() -> List[OperatorConfig]:
+    supported_config_and_operators: List[OperatorConfig] = []
+    for quantization_config in [get_default_x86_inductor_quantization_config(), ]:
+        ops = supported_quantized_operators()
+        for op_string, pattern_list in ops.items():
+            supported_config_and_operators.append(
+                OperatorConfig(quantization_config, pattern_list)
+            )
+    return copy.deepcopy(supported_config_and_operators)
+
+
+@functools.lru_cache
+def get_default_x86_inductor_quantization_config():
```

Comment:
So what is really different in this quantizer? Is it qspec?

Reply:
Did you mean the difference between `X86InductorQuantizer` and `QNNPackQuantizer`? If so, I think: 1. the supported fusion patterns are different. 2. We will enable some quantization propagation instead of just pattern match for the quantization recipes.

Reply:
Sure, in general, after we annotate the fusion patterns of conv/linear, we will go through the graph from start to end and checking if a single quantizable node (for example: relu) is connected to a quantized node of not. If so, we can also annotate this quantizable node as quantized node. We can share more details offline if you are interested.

Reply:
> quantized node

Yes I am interested in understanding this a bit better. If you can share more details that would be great. @jerryzh168 whats the best way to do this? If you are allowed to share here, thats even better since we dont have to find a separate communication channel

Reply:
@kimishpatel I posted their IPEX recipe internally in the working group, see the end of the slides